File: Number_of_Laser_Beams_in_a_Bank.py
# Title: 2125. Number of Laser Beams in a Bank
# Difficulty: Medium
# Problem: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/description

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        prev, count = 0, 0

        for i in range(len(bank)):
            temp = bank[i].count('1')
            count += prev * temp
            if temp:
                prev = temp
        
        return count

File: Number_of_Laser_Beams_in_a_Bank.cpp
// TItle: 2125. Number of Laser Beams in a Bank
// Difficulty: Medium
// Problem: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/description

class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int prev=0, count=0;

        for(string row : bank){
            int ones=0;
            for(char bit : row){
                if (bit == '1'){
                    ones++;
                }
            }
            count += ones * prev;
            if (ones)
            {
                prev = ones;
            }
        }
        return count;
    }
};
